wolfcrypt/src/asn.c

```diff
@@ -10555,7 +10555,10 @@ int PemToDer(const unsigned char* buff, long longSz, int type,
 #ifndef NO_WOLFSSL_SKIP_TRAILING_PAD
                 #ifndef NO_DES3
                     if (info->cipherType == WC_CIPHER_DES3) {
-                        padVal = der->buffer[der->length-1];
+                        if (der->length > 0)
+                            padVal = der->buffer[der->length-1];
+                        else
+                            padVal = WC_CIPHER_NONE;
```

Comment:
It's not correct to set padVal to something and then adjust the length on the next line based on the value.  Even if the value is now 0 that may change.

Reply:
Added more stringent checks for

A) is padVal or der->length 0? If not do next check
C) is padVal <= DES_BLOCK_SIZE? If so do next check
D) is der->length sufficient to reduce by padVal? If so reduce.

if ((padVal | der->length) != 0 && padVal <= DES_BLOCK_SIZE
                              && der->length > DES_BLOCK_SIZE) {